pkg/filestore/s3/s3.go

```diff
@@ -84,55 +79,47 @@ func NewStore(region, bucket string, opts ...Option) (*Store, error) {
+			v.RoleARN = s.roleARN
+			v.TokenRetriever = stscreds.IdentityTokenFile(s.tokenFile)
+		}))
+	}
+
+	// When you initialize an aws.Config instance using config.LoadDefaultConfig, the SDK uses its default credential chain to find AWS credentials.
+	// This default credential chain looks for credentials in the following order:
+	//
+	// 1. Environment variables.
+	//   1. Static Credentials (AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY, AWS_SESSION_TOKEN)
+	//   2. Web Identity Token (AWS_WEB_IDENTITY_TOKEN_FILE)
+	// 2. Shared configuration files.
+	//   1. SDK defaults to credentials file under .aws folder that is placed in the home folder on your computer.
+	//   2. SDK defaults to config file under .aws folder that is placed in the home folder on your computer.
+	// 3. If your application uses an ECS task definition or RunTask API operation, IAM role for tasks.
+	// 4. If your application is running on an Amazon EC2 instance, IAM role for Amazon EC2.
+	// ref: https://aws.github.io/aws-sdk-go-v2/docs/configuring-sdk/#specifying-credentials
+	cfg, err := config.LoadDefaultConfig(ctx, optFns...)
 	if err != nil {
-		return nil, fmt.Errorf("failed to create a session: %w", err)
-	}
-
-	creds := credentials.NewChainCredentials(
-		[]credentials.Provider{
-			&credentials.EnvProvider{},
-			&credentials.SharedCredentialsProvider{
-				Filename: s.credentialsFile,
-				Profile:  s.profile,
-			},
-			// roleSessionName specifies the IAM role session name to use when assuming a role.
-			// it will be generated automatically in case of empty string passed.
-			// ref: https://github.com/aws/aws-sdk-go/blob/0dd12669013412980b665d4f6e2947d57b1cd062/aws/credentials/stscreds/web_identity_provider.go#L116-L121
-			stscreds.NewWebIdentityRoleProvider(sts.New(sess), s.roleARN, "", s.tokenFile),
-			&ec2rolecreds.EC2RoleProvider{
-				Client: ec2metadata.New(sess),
-			},
-		},
-	)
-	cfg := aws.NewConfig().WithRegion(region).WithCredentials(creds)
-	s.client = s3.New(sess, cfg)
+		return nil, fmt.Errorf("failed to load config to create s3 client: %w", err)
+	}
+	s.client = s3.NewFromConfig(cfg)
 
 	return s, nil
 }
 
-func (s *Store) NewReader(ctx context.Context, path string) (rc io.ReadCloser, err error) {
+func (s *Store) NewReader(ctx context.Context, path string) (io.ReadCloser, error) {
 	input := &s3.GetObjectInput{
 		Bucket: aws.String(s.bucket),
 		Key:    aws.String(path),
 	}
-	out, err := s.client.GetObjectWithContext(ctx, input)
+	out, err := s.client.GetObject(ctx, input)
 	if err != nil {
-		if aerr, ok := err.(awserr.Error); ok {
-			switch aerr.Code() {
-			case s3.ErrCodeNoSuchKey:
-				err = filestore.ErrNotFound
-				return
-			case s3.ErrCodeInvalidObjectState:
-				err = fmt.Errorf("invalid object state: %w", err)
-				return
-			default:
-				err = fmt.Errorf("unexpected aws error given: %w", err)
-				return
-			}
-		}
-		err = fmt.Errorf("unknown error given: %w", err)
-		return
```

Comment:
Why was this part removed?

Reply:
Nice catch 👍 I missed handling in case not found s3 object 🙏

Reply:
addressed on https://github.com/pipe-cd/pipe/pull/1671/commits/5a02024e8edf63536d6edb302993b4d8495f694a 🙏